[Docs][ML][SIEM] Combines prebuilt job descriptions (#932)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/ootb-ml-jobs.asciidoc

[[ootb-ml-jobs-siem]]
=== SIEM

These {anomaly-jobs} appear by default in the Anomaly Detection interface of
the {siem-guide}/machine-learning.html[SIEM app] in {kib}. They help you 

> automatically detect file system and network anomalies on your hosts.

Author: @szabosteve @lcawl 
Don't know if you want to add some info here as to why the ECS fields are listed for each job?

Reviewer: @benskelker @lcawl 
Sure, I'll add in a separate PR that when you are not using Beats, you need to map your data to the respective ECS fields.

Reviewer: @benskelker @lcawl 
I added a sentence about the listed ECS fields as a single comment to avoid conflicts. Please let me know if you would modify or rather omit it.

Author: Thanks @szabosteve

> automatically detect file system and network anomalies on your hosts.

Reviewer: ```suggestion
automatically detect file system and network anomalies on your hosts. However,
if you don't use Beats, you need to map your data to the ECS fields that are 
listed for every job.
```

processes usually only run occasionally on a single machine but are common to
all or many hosts in a fleet.

Job details:::

* Analyzes host activity logs where `agent.type` is `auditbeat` (Linux) or `winlogbeat` (Windows).
* Models the occurrences of processes on all hosts.
* Detects processes that occur rarely compared to other processes on all 
  hosts (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).

Required {beats}:::

* {auditbeat} (Linux)
* {winlogbeat} (Windows)

Required ECS fields when not using {beats}:::

* `host.name`
* `process.name`
* `user.name`
* `process.executable`
* `event.action`
* `agent.type`

linux_anomalous_user_name_ecs::
+
windows_anomalous_user_name_ecs::

Searches for activity from users who are not normally active, which can
indicate unauthorized changes, activity by unauthorized users, lateral
movement, and compromised credentials.
+
In organizations, new usernames are not often created apart from specific types 
of system activities, such as creating new accounts for new employees. These 
user accounts quickly become active and routine.
+
Events from rarely used usernames can point to suspicious activity. 

> Additionally, automated Linux fleets tent to see activity from rarely used 

Reviewer 2: ```suggestion
Additionally, automated Linux fleets tend to see activity from rarely used 
```

scanners.

Job details:::

* Analyzes host activity logs where `agent.type` is `winlogbeat`.
* Models occurrences of process creation activities (`partition_field_name` is 
  `process.parent.name`).
* Detects process relationships that are rare compared to other process 
  relationships (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).

Required {beats}:::

* {winlogbeat} (Windows)

Required ECS fields when not using {beats}:::

* `host.name`
* `process.name`
* `user.name`
* `process.parent.name`
* `event.action`
* `agent.type`

windows_anomalous_script::

Searches for PowerShell scripts with unusual data characteristics, such as
obfuscation, that may be a characteristic of malicious PowerShell script text
blocks.

Job details:::

* Analyzes host activity logs where `agent.type` is `winlogbeat`.
* Models occurrences of PowerShell script activities.
* Detects unusual PowerShell script execution compared to other PowerShell 
  script activities (using the 
  {ml-docs}/ml-info-functions.html#ml-info-content[`high_info_content` function]).

Required {beats}:::

* {winlogbeat} (Windows)

+

> NOTE: You can only run this job when Winlogbeat is used to ship data.

Reviewer 2: Attempting to clarify whether there is a failure to run (i.e. job is visible but fails) or whether you never see this job:
```suggestion
NOTE: This job is available only when you use {winlogbeat} to ship data.
```

Author: Yep - will update all the relevant notes.

These {anomaly-jobs} appear by default in the Anomaly Detection interface of the 
{siem-guide}/machine-learning.html[SIEM app] in {kib}. They help you 
automatically detect file system and network anomalies on your hosts.

// tag::siem-jobs[]
linux_anomalous_network_activity_ecs::
+
windows_anomalous_network_activity_ecs::

> Identifies Windows/Linux processes that do not usually use the network but have

Reviewer 3: ```suggestion
Identifies processes that do not usually use the network but have
```
This is maybe more style, but I think this makes the sentence easier to read. The OS specifiers in the names and in the required query I think clearly identify what OS the process is running on.

Author: Sure, but we'll wait for @randomuserid as the descriptions are basically his.

unexpected network activity, which can indicate command-and-control, lateral
movement, persistence, or data exfiltration activity.
+
A process with unusual network activity can denote process exploitation or
injection, where the process is used to run persistence mechanisms that allow a
malicious actor remote access or control of the host, data exfiltration, and
execution of unauthorized network applications.

Job details:::

* Analyzes network activity logs where `agent.type` is `auditbeat` (Linux) or `winlogbeat` (Windows).
* Models the occurrences of processes that cause network activity.
* Detects network activity caused by processes that occur rarely compared to 
  other processes (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).

> * {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Reviewer 2: In my opinion, these repeated links to info about bucket spans and influencers are distracting.  I'd suggest having them appear once at the top of the page.

Reviewer 3: I agree with this. Maybe before the first job have a short description of what each of the terms mean  (single sentence summary) with the existing hyperlink for more detailed info. And maybe have a single list of the jobs at the top of the page with links to the job itself. I'm thinking like the [Logstash mutate filter](https://www.elastic.co/guide/en/logstash/current/plugins-filters-mutate.html#plugins-filters-mutate-options)

Author: I think a lot of people will be using ctrl/cmd f on this page, searching for a specific job.
I don't mind though.

Reviewer 2: @benskelker Per our chat, I think the bucket span and influencer details can be omitted from this list (and therefore the links to bucket span and influencer definitions too). One thing we might consider adding to all these pages, however, is a link to where the modules are defined: https://github.com/elastic/kibana/tree/master/x-pack/plugins/ml/server/models/data_recognizer/modules

Then folks can dig into the exact job and datafeed definitions if they so desire.  For the SIEM auditbeat ones can be found here: https://github.com/elastic/kibana/tree/master/x-pack/plugins/ml/server/models/data_recognizer/modules/siem_auditbeat/ml

Reviewer 2: I've merged that change in https://github.com/elastic/stack-docs/pull/933. If you want help with the merge conflict, please just let me know!

Beats required on hosts:::

* Auditbeat (Linux)
* Winlogbeat (Windows)

Required ECS fields:::

* `destination.ip` ({ml-docs}/ml-influencers.html[influencer])
* `host.name` (influencer)
* `process.name` (influencer)
* `user.name` (influencer)
* `event.action`
* `agent.type`

linux_anomalous_network_port_activity_ecs::

Identifies unusual destination port activity that can indicate
command-and-control, persistence mechanism, or data exfiltration activity.
+
Rarely used destination port activity is generally unusual in Linux fleets, and 
can indicate unauthorized access or threat actor activity.

Job details:::

* Analyzes network activity logs where `agent.type` is `auditbeat`.
* Models destination port activity.
* Detects destination port activity that occurs rarely compared to other port 
  activities (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).
* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

+
Beats required on hosts:::

* Auditbeat (Linux)

+
NOTE: You can only run this job when Auditbeat is used to ship data.

> {ml-docs}/ml-influencers.html[Influencers]:::

Reviewer 2: Shouldn't this match the other sections and have this title and then "(influencer)" after each field?
```suggestion
Required ECS fields:::
```

Author: @lcawl  
hmm not sure here. From our perspective, it's only interesting when users are not using beats and can manually map their data to ECS. When a job can be mapped, the influencers are always ECS fields. When a job cannot be mapped because it relies on non-ECS beat fields, I listed the influencers separately.

What do you think?

Reviewer 3: Are the non-influencer fields "detector" fields? If I was reading this documentation for the first time trying to understand the ML job options, it's not clear to me what these other fields are doing.

Reviewer 2: Sorry, I'm not sure I understand the difference.  These look like the same field names that exist in other "ECS fields" sections, so if they're not the same I think further clarification is required.

Reviewer 3: The way I read the description, the `agent.type` must be as specified in these jobs, so ECS-compliant data from other sources wouldn't work. Is that accurate?

Author: This is the intro sentence in the SIEM docs, before including your stuff:

If you use Beats to ship your data, the ECS fields are populated and you 
do not need to manually map the fields.

Not sure, but maybe add something to your file before the include?

Author: > The way I read the description, the `agent.type` must be as specified in these jobs, so ECS-compliant data from other sources wouldn't work. Is that accurate?

Don't think so. The `agent.type` would need to be mapped to the user's data shipper. The issue with some jobs is that they rely on non-compliant ECS fields, so the only way they can be run is when beats are used to ship the data.

* `destination.ip`
* `host.name` 
* `process.name`
* `user.name`

linux_anomalous_network_service::

Searches for unusual listening ports that can indicate execution of
unauthorized services, backdoors, or persistence mechanisms.

Job details:::

* Analyzes network activity logs where `agent.type` is `auditbeat`.
* Models listening port activity.
* Detects listening port activity that occurs rarely compared to 
  other port activities (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).
* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Beats required on hosts:::

* Auditbeat (Linux)

+
NOTE: You can only run this job when Auditbeat is used to ship data.

> {ml-docs}/ml-influencers.html[Influencers]:::

Reviewer 2: Ditto re changing title and adding "(influencer)" after each field:
```suggestion
Required ECS fields:::
```

* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Beats required on hosts:::

* Winlogbeat (Windows)

+
NOTE: You can only run this job when Winlogbeat is used to ship data.

{ml-docs}/ml-influencers.html[Influencers]:::

* `host.name` 
* `user.name`
* `winlog.event_data.Path`

windows_anomalous_service::

Searches for unusual Windows services that can indicate execution of
unauthorized services, malware, or persistence mechanisms.
+
In corporate Windows environments, hosts do not generally run many rare or
unique services. This job helps detect malware and persistence mechanisms that
have been installed and run as a service.

Job details:::

* Analyzes host activity logs where `agent.type` is `winlogbeat`.
* Models occurrences of Windows service activities.
* Detects Windows service activities that occur rarely compared to other Windows service activities (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).
* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Beats required on hosts:::

* Winlogbeat (Windows)

+
NOTE: You can only run this job when Winlogbeat is used to ship data.

> {ml-docs}/ml-influencers.html[Influencers]:::

Reviewer 2: Ditto
```suggestion
Required ECS fields:::
```

* Detects user context switches that occur rarely compared to other user context switches (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).
* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Beats required on hosts:::

* Winlogbeat (Windows)

Required ECS fields:::

* `process.name` ({ml-docs}/ml-influencers.html[influencer])
* `host.name` (influencer)
* `user.name` (influencer)
* `event.code`
* `agent.type`

windows_rare_user_type10_remote_login::

Searches for unusual remote desktop protocol (RDP) logins, which can indicate
account takeover or credentialed persistence using compromised accounts. RDP
attacks, such as BlueKeep, also tend to use unusual usernames.

Job details:::

* Analyzes host activity logs where `agent.type` is `winlogbeat`.
* Models occurrences of user remote login activities.
* Detects user remote login activities that occur rarely compared to other 
  user remote login activities (using the {ml-docs}/ml-rare-functions.html#ml-rare[`rare` function]).
* {ml-docs}/ml-buckets.html[Bucket span]: 15 minutes.

Beats required on hosts:::

* Winlogbeat (Windows)

+
NOTE: You can only run this job when Winlogbeat is used to ship data.

> {ml-docs}/ml-influencers.html[Influencers]:::

Reviewer 2: Ditto:
```suggestion
Required ECS fields:::
```
